### DISCUSSION TITLE
What happens when a message cannot be delivered to AWS EventBridge.

### DISCUSSION THREAD

**Discussion by robvanpamel** [aws/aws-dotnet-messaging#129] [Q&A] [answered]
Looking at the EventBridgePublisher, I notice that the normal PutEvents functionality is used. (https://github.com/awslabs/aws-dotnet-messaging/blob/main/src/AWS.Messaging/Publishers/EventBridge/EventBridgePublisher.csl) 

However, I learned that the provided response can be successful although that some of the messages failed. This is why we should look at the response and look at the entries to identify which ones are successful. Is there a reason this response is not being passed through and a void (Task) is used? If not, should we create a PR to improve this? 

See https://docs.aws.amazon.com/sdkfornet/v3/apidocs/items/EventBridge/TPutEventsResultEntry.html   https://docs.aws.amazon.com/sdkfornet/v3/apidocs/items/EventBridge/TPutEventsResponse.html

**Comment by normj** [comment]
That is a miss on our side. We should be communicating back to the caller the failure. Happy to take a PR if you are interested in contributing that.

**Comment by robvanpamel** [comment, marked as answer]
I created the PR here 
https://github.com/awslabs/aws-dotnet-messaging/pull/130 
Let me know what you think. Due to the abstractions, I had to take some shortcuts where I am not 100% convinced from (eg naming)
